### PR TITLE
initial "Importer" implementation for datacenter and network resources

### PR DESCRIPTION
**Pull request by bmanojlovic**
This is work in progress but in current implementation it is "good" enough that it does allow you to create VM on ovirt/RHV environments with existing datacenter or networks that could be imported
Currently it works like this:
in ovirt-shell you look up id of datacenter and do import like this
```
steki@pc:~/k8s-dxc> TF_LOG=TRACE TF_LOG_PATH=terraform.log terraform import -config=. -provider=ovirt "ovirt_datacenter.my_dc" 68e4fc08-4815-408f-a03a-4ea187aee1a5
ovirt_datacenter.my_dc: Importing from ID "68e4fc08-4815-408f-a03a-4ea187aee1a5"...
ovirt_datacenter.my_dc: Import complete!
  Imported ovirt_datacenter (ID: 68e4fc08-4815-408f-a03a-4ea187aee1a5)
ovirt_datacenter.my_dc: Refreshing state... (ID: 68e4fc08-4815-408f-a03a-4ea187aee1a5)

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

steki@pc:~/k8s-dxc>
```
of course same logic for network import in "simple" resource imports.
Please comment first do not accept it as is, as I am still working on it.

example terraform plan

```
steki@pc:~/k8s-dxc> TF_LOG=TRACE TF_LOG_PATH=terraform.log terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

ovirt_datacenter.my_dc: Refreshing state... (ID: 68e4fc08-4815-408f-a03a-4ea187aee1a5)
data.ovirt_datacenters.defaultDC: Refreshing state...
ovirt_network.TESTV_222: Refreshing state... (ID: 9a34ec0e-b433-4adb-9582-1ade9f51a9d8)

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + ovirt_disk.my_disk_1
      id:                              <computed>
      alias:                           "my_disk_1"
      format:                          "cow"
      name:                            "my_disk_1"
      size:                            "23687091200"
      sparse:                          "true"
      storage_domain_id:               "75832078-143c-4720-b704-09795a9bb7ca"

  + ovirt_disk_attachment.my_diskattachment_1
      id:                              <computed>
      active:                          "true"
      bootable:                        "false"
      disk_id:                         "${ovirt_disk.my_disk_1.id}"
      interface:                       "virtio"
      read_only:                       "false"
      use_scsi_reservation:            "false"
      vm_id:                           "${ovirt_vm.my_vm_1.id}"

  + ovirt_vm.my_vm_1
      id:                              <computed>
      authorized_ssh_key:              "ssh-rsa AAAAB3..."
      cluster:                         "My Cluster"
      cores:                           "1"
      name:                            "my_vm_1"
      network_interface.#:             "1"
      network_interface.0.boot_proto:  "static"
      network_interface.0.gateway:     "10.1.17.254"
      network_interface.0.ip_address:  "10.1.16.250"
      network_interface.0.label:       "eth0"
      network_interface.0.on_boot:     "true"
      network_interface.0.subnet_mask: "255.255.254.0"
      sockets:                         "1"
      template:                        "Blank"
      threads:                         "1"


Plan: 3 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.

steki@pc:~/k8s-dxc> 

```

### CODE REVIEW

**Comment by imjoey** [comment]
Nice work, Thanks!  I saw you've explicitly defined two functions to do resources importing. While what do you think of choosing the typical `schema.ImportStatePassthrough` as the `Importer` function, which is simply filled with the given `id` and then would read/update all attributes after running `terraform plan`.

File: ovirt/resource_ovirt_datacenter.go [hunk 1]
```diff
@@ -80,7 +83,7 @@ func resourceOvirtDataCenterUpdate(d *schema.ResourceData, meta interface{}) err
 			datacenterBuilder.Name(name.(string))
 		}
 	} else {
-		return fmt.Errorf("DataCenter's name does not exist!")
+		return fmt.Errorf("datacenter's name does not exist")
```
Comment (imjoey) [inline ovirt/resource_ovirt_datacenter.go]: Although the capitalized starting word in error string is unidiomatic, the terraform official providers all do so, which I think is a kind of following the rule of terraform log `ovirt_datacenter.my_dc: Importing from ID "68e4fc08-4815-408f-a03a-4ea187aee1a5"`. Please refer to [AWS error string](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_instance.go#L572).

Comment (bmanojlovic) [reply ovirt/resource_ovirt_datacenter.go]: regarding capitalized words in error format is my mistake that should not go into this PR, sorry will revert back.

**Comment by bmanojlovic** [comment]
What i found strange was diskattachment ID , is there some specific reason it is not real ID from api?
Instead in current incarnation it is `VM_ID:DISK_ID` which is not real issue but a bit confusing :)
(btw i have reverted change of that error message)

**Comment by imjoey** [comment]
This is a kind of compromise. According to the official doc [Get a Disk Attachment](http://ovirt.github.io/ovirt-engine-api-model/4.3/#services/disk_attachment/methods/get), besides providing the `disk-attachment-ID`, a `vm-ID` is also required when retrieving a disk attachment object. So the `vm-ID` must be added as part of the terraform resource ID. 

After referring to the codes (DiskAttachmentMapper.java at ovirt-engine)[https://github.com/oVirt/ovirt-engine/blob/68753f46f09419ddcdbb632453501273697d1a20/backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/types/DiskAttachmentMapper.java], I found that it exactly takes the `disk-ID` as the ID of newly added disk attachment. That's why I pass a `disk-ID` to locate the `diskattachmentService` in the code blow:

```go
attachmentService := conn.SystemService().
		VmsService().
		VmService(vmID).
		DiskAttachmentsService().AttachmentService(diskID)
```

**Comment by bmanojlovic** [comment]
of course will remove them, still working on it anyway :)

**Comment by bmanojlovic** [comment]
removed function, added ovirt_networks data provider - still in progress ;)

**Comment by bmanojlovic** [comment]
Current version creates VM with correct network attached (existing from data provider)
Missing features are of course storage_domain_id and Datacenter Cluster information which is currently hard-coded into main.tf and can not be fetched from ovirt api
(all tests i am performing against RHV 4.1.10.3-0.1.el7 )

**Comment by bmanojlovic** [comment]
state of this  is "works for me", @imjoey code changes are done as requested.

File: ovirt/resource_ovirt_vm.go [hunk 2]
```diff
@@ -168,7 +175,42 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 	if ok {
 		d.SetId(newVM.MustId())
 	}
-
+	for i := 0; i < numNetworks; i++ {
+		prefix := fmt.Sprintf("network_interface.%d", i)
+		profilesService := conn.SystemService().VnicProfilesService()
+		var profileID, pfDcID, newVMDcID string
+		pfsResp, _ := profilesService.List().Send()
```
Comment (imjoey) [inline ovirt/resource_ovirt_vm.go]: `MustSend()` could be used to ignore the error returned.

**Comment by imjoey** [comment]
@bmanojlovic  Great thanks for your work. 😄

**Comment by imjoey** [comment]
@bmanojlovic  Sorry for my mistake on merging. I just rebased and squashed the commits.

**Pull request merged**